packages/pretty-format/src/plugins/immutable.js

```diff
@@ -0,0 +1,195 @@
+
+const IS_LIST_SENTINEL = '@@__IMMUTABLE_LIST__@@';
+const IS_MAP_SENTINEL = '@@__IMMUTABLE_MAP__@@';
+const IS_SEQ_SENTINEL = '@@__IMMUTABLE_SEQ__@@';
+const IS_SET_SENTINEL = '@@__IMMUTABLE_SET__@@';
+const IS_STACK_SENTINEL = '@@__IMMUTABLE_STACK__@@';
+
+const IS_ORDERED_SENTINEL = '@@__IMMUTABLE_ORDERED__@@';
+
+const getName = name => 'Immutable.' + name;
+const SPACE = ' ';
+
+const printImmutableEntries = (
+  val: any,
+  config: Config,
+  indentation: string,
+  depth: number,
+  refs: Refs,
+  printer: Printer,
+  type: string,
+): string =>
+  ++depth > config.maxDepth
+    ? '[' + getName(type) + ']'
+    : getName(type) +
+      SPACE +
+      '{' +
+      (val.size !== 0
+        ? printIteratorEntries(
+            val.entries(),
+            config,
+            indentation,
+            depth,
+            refs,
+            printer,
+          )
+        : config.spacingOuter + indentation) +
+      '}';
+
+// Return an iterator for Immutable Record in v4 or later.
+const getRecordEntries = val => {
+  let i = 0;
+  return {
+    next() {
+      if (i < val._keys.length) {
+        const key = val._keys[i++];
+        return {done: false, value: [key, val.get(key)]};
+      }
+      return {done: true};
+    },
+  };
+};
+
+const printImmutableRecord = (
+  val: any,
+  config: Config,
+  indentation: string,
+  depth: number,
+  refs: Refs,
+  printer: Printer,
+): string => {
+  // _name property is defined only for an Immutable Record instance
+  // which was constructed with a second optional descriptive name arg
+  const name = getName(val._name || 'Record');
+  const size = Array.isArray(val._keys) ? val._keys.length : val.size;
+  const entries = typeof Array.isArray(val._keys)
+    ? getRecordEntries(val) // v4 or later
+    : val.entries(); // v3 or earlier
+  return ++depth > config.maxDepth
+    ? '[' + name + ']'
+    : name +
+      SPACE +
+      '{' +
+      (size !== 0
+        ? printIteratorEntries(
+            entries,
+            config,
+            indentation,
+            depth,
+            refs,
+            printer,
+          )
+        : config.spacingOuter + indentation) +
+      '}';
+};
+
+const printImmutableValues = (
+  val: any,
+  config: Config,
+  indentation: string,
+  depth: number,
+  refs: Refs,
+  printer: Printer,
+  type: string,
+): string =>
+  ++depth > config.maxDepth
+    ? '[' + getName(type) + ']'
+    : getName(type) +
+      SPACE +
+      '[' +
+      (val.size !== 0
+        ? printIteratorValues(
+            val.values(),
+            config,
+            indentation,
+            depth,
+            refs,
+            printer,
+          )
+        : config.spacingOuter + indentation) +
+      ']';
+
+export const serialize = (
+  val: any,
+  config: Config,
+  indentation: string,
+  depth: number,
+  refs: Refs,
+  printer: Printer,
+): string => {
+  if (val[IS_SEQ_SENTINEL]) {
+    return '[' + getName('Seq') + ']';
```

Comment:
Because `Record` doesn’t have a sentinel in immutable v3, `serialize` needs to account for all types that do. The unified `test` matches `Seq` too, which seems easiest to understand. Do you mean support it better?